networkx/drawing/nx_pylab.py

```diff
@@ -561,11 +736,12 @@ def draw_networkx_edges(
         width. See `matplotlib.patches.FancyArrowPatch` for attribute
         `mutation_scale` for more info.
 
-    connectionstyle : string (default="arc3")
+    connectionstyle : string, Iterable (default="arc3")
         Pass the connectionstyle parameter to create curved arc of rounding
         radius rad. For example, connectionstyle='arc3,rad=0.2'.
         See `matplotlib.patches.ConnectionStyle` and
         `matplotlib.patches.FancyArrowPatch` for more info.
+        If Iterable, index indicates i'th edge key of MultiGraph
```

Comment:
Let's make this more clear in terms of edge key.  It is not the edge key as stored in the MultiGraph. The index matches the index of the edges in the input `edgelist`.  Right?  If so, maybe
```
    If iterable, the connectionstyle order matches the edge order in `edgelist`
```

Reply:
It is actually edge key, as in (u, v, edge_key/edge_id). There was a single connectionstyle argument for all lines before, so this is just an extension when there is more than 1 connection between 2 nodes.

Reply:
So how does the iterable work when one node-pair has 2 edges and another node pair has 5 edges? 
And what if the node-pair has more edges than the iterable contains?
Would it be better to have the iterable match the edgelist order? What's the advantage of this approach?

Reply:
It takes the first 2 items for the former and first 5 for the latter.

I have not actually thought about it. All the work that has been done prior took this approach.

If the input is shorter than the number of edges of a node pair, it will result in an IndexError. I thought about making a pre-validation, but thought it might be too expensive to do it. But now I think it is a fairly quick operation.

So I can do this pre-check.

But the question is if it is the desired approach. From my POV, I would rather have simpler input in this instance as opposed to micro-management. If I know that my graph has max 2 directed edges, I can give 0.1 and 0.2 rad and I know they will not overlap. If I have arbitrary amount of multi-edges, I can just make infinite iterator which accumulates 0.1 rad so that I have 0.1, 0.2, etc... Constructing an input manually for each edge does seem a bit of work. So simplicity of use is the benefit here. Although I see the benefit of being able to set it manually for each edge. This way, some space-saving and nicer layouts are possible, but I am not sure how much this would be used.

However, maybe I am missing something.

What I could do is add an extra flag which indicates if the iterator is along `edge key` or `edgelist` dimension.

Reply:
I think you are right. It is simpler this way.  
Let's leave it the way it is (since it is the approach always used before).
We can think about it and add a new approach if and when a better approach becomes more clear.
Thanks!